Tasks/AzureKeyVault/models/KeyVaultTaskParameters.ts

```diff
@@ -18,6 +19,7 @@ export class KeyVaultTaskParameters {
         this.subscriptionId = tl.getEndpointDataParameter(connectedService, "SubscriptionId", true);
         this.keyVaultName = tl.getInput("KeyVaultName", true);
         this.secretsFilter = tl.getDelimitedInput("SecretsFilter", ",", true);
+        this.flattenVariableName = tl.getInput("FlattenVariableName", true);
```

Comment:
Rename FlattenVariableName to SYSTEM_FLATTENED_SECRET_NAME OR SYSTEM_FLATTENEDSECRETNAME

Reply:
Could you please clarify why we should use that name? We'd like to introduce a new GUI parameter here, not a *magic* variable.

Reply:
@lovakumar could you please have a look?

Reply:
@cervengoc if you want to add it to GUI as parameter then it is fine. We don't need this magic variable.

Though FlattenVariableName is optional and it is 90% scenario that it will be blank. 
@RoopeshNair can you please comment on the spec here.